voc/python/ast.py

```diff
@@ -1747,18 +1748,39 @@ def visit_Yield(self, node):
 
     @node_visitor
     def visit_YieldFrom(self, node):
-        # expr value):
-        print(
-            '"yield from" statement not yet implemented. '
-            'Will throw a NotImplementedError at runtime'
+        self.visit(node.value)  # pushes expression on stack
+        self.context.add_opcodes(
+            python.Object.iter()  # pops expression from stack then gets and pushes its iterator on stack
         )
+        self.context.store_name('#yield-iter-%x' % id(node))
+
+        loop = START_LOOP()
         self.context.add_opcodes(
-            java.THROW(
-                'org/python/exceptions/NotImplementedError',
-                ['Ljava/lang/String;',
-                 JavaOpcodes.LDC_W('"yield from" statement not yet implemented')]
-            )
+            loop,
+        )
+        self.context.add_opcodes(
+            TRY(),
+        )
```

Comment:
As a style thing, it's preferable to include as many opcodes as possible in a single `add_opcodes` call. The end result is exactly the same in terms of functionality, but there's less code to *read*, which can make the intent of the code easier to interpret for those coming in later.